driver-core/src/main/com/mongodb/internal/operation/CommandOperationHelper.java

```diff
@@ -412,9 +419,8 @@ static <T, R> R executeRetryableWrite(
                     // attach `maxWireVersion`, `retryableCommandFlag` ASAP because they are used to check whether we should retry
                     retryState.attach(AttachmentKeys.maxWireVersion(), maxWireVersion, true)
                             .attach(AttachmentKeys.retryableCommandFlag(), isRetryWritesEnabled(command), true)
-                            .attach(AttachmentKeys.commandDescriptionSupplier(), command::getFirstKey, true)
+                            .attach(AttachmentKeys.commandDescriptionSupplier(), command::getFirstKey, false)
                             .attach(AttachmentKeys.command(), command, false);
```

Comment:
I don't know why the command description for writes was auto-cleared, but given that the command was not auto-cleared, auto-clearing the description did not make sense.